crates/web-client/src/models/account_component.rs

```diff
@@ -68,13 +70,24 @@ impl AccountComponent {
     }
 
     #[wasm_bindgen(js_name = "createAuthComponent")]
-    pub fn create_auth_component(secret_key: &SecretKey) -> AccountComponent {
+    pub fn create_auth_component(secret_key: &SecretKey) -> Result<AccountComponent, JsValue> {
         let native_secret_key: NativeSecretKey = secret_key.into();
-        let native_auth_component: NativeAccountComponent = NativeRpoFalcon512::new(
-            PublicKeyCommitment::from(native_secret_key.public_key().to_commitment()),
-        )
-        .into();
-        AccountComponent(native_auth_component)
+        match native_secret_key {
+            NativeSecretKey::EcdsaK256Keccak(_) => Err(JsValue::from_str(
+                "building auth component from an ecdsa key is not yet soported",
```

Comment:
nit: rest of `JsValue::from_str` on the codebase begin with uppercase.
```suggestion
                "Building auth component from an ecdsa key is not yet soported",
```

Reply:
I forgot to tackle this! I agree we should do it consistently